gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.ts

```diff
@@ -0,0 +1,317 @@
+        {
+          width: GIO_DIALOG_WIDTH.SMALL,
+          data: { name: '', value: '' },
+          role: 'alertdialog',
+          id: 'createHeaderDialog',
+        },
+      )
+      .afterClosed()
+      .pipe(
+        filter((data): boolean => !!data),
+        switchMap((data: ApiPortalHeaderEditDialogResult) => this.environmentApiHeadersService.createApiHeader(data)),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  public updateHeader(header: ApiPortalHeader): void {
+    this.matDialog
+      .open<ApiPortalHeaderEditDialogComponent, ApiPortalHeaderDialogData, ApiPortalHeaderEditDialogResult>(
+        ApiPortalHeaderEditDialogComponent,
+        {
+          data: { name: header.name, value: header.value },
+          width: GIO_DIALOG_WIDTH.SMALL,
+        },
+      )
+      .afterClosed()
+      .pipe(
+        filter((data): boolean => !!data),
+        switchMap((headerDialogResult: ApiPortalHeaderEditDialogResult) => {
+          const payload: ApiPortalHeader = {
+            ...header,
+            name: headerDialogResult.name,
+            value: headerDialogResult.value,
+          };
+          return this.environmentApiHeadersService.updateApiHeader(payload);
+        }),
+        tap(() => {
+          this.snackBarService.success('API Information updated successfully');
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  public deleteHeader(header: ApiPortalHeader): void {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data: {
+          title: 'Delete API Information',
+          content: 'Are you sure you want to delete this API Information?',
+          confirmButton: 'Delete',
+        },
+        role: 'alertdialog',
+        id: 'deletePortalHeaderConfirmDialog',
+      })
+      .afterClosed()
+      .pipe(
+        filter((confirm: boolean): boolean => confirm),
+        switchMap(() => this.environmentApiHeadersService.deleteApiHeader(header)),
+        tap(() => {
+          this.snackBarService.success(`API Information ${header.name} deleted successfully`);
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  private initializeForm() {
+    this.form = new FormGroup<ApiForm>({
+      apiKeyHeader: new FormControl<string>(this.settings.portal.apikeyHeader),
+    });
+    this.formInitialValues = this.form.getRawValue();
+
+    const isReadOnly = !this.permissionService.hasAnyMatching(['environment-api-u']);
+    if (isReadOnly) {
+      this.form.disable();
+    }
+  }
+
+  reset() {
+    this.form.reset();
+    this.initializeForm();
```

Comment:
Is it necessary to call `this.initializeForm()`? I think the `.reset()` you call above should do the trick?

Reply:
No, it does not work properly without it.
If removed - when discard is clicked - the whole value of api-key-header is removed.
We need to refresh page and involve this.initializeForm(); that is in the `ngOnInit` to have it back.